Source/WebCore/workers/service/server/SWServerRegistration.cpp

```diff
@@ -423,4 +424,39 @@ std::optional<ExceptionData> SWServerRegistration::setNavigationPreloadHeaderVal
     return { };
 }
 
+void SWServerRegistration::addCookieChangeSubscriptions(Vector<CookieChangeSubscription>&& subscriptions)
+{
+    for (auto& requestedSubscription : subscriptions) {
+        bool isNewSubscription = true;
+        for (auto& currentSubscription : m_cookieChangeSubscriptions) {
+            if (currentSubscription.name == requestedSubscription.name && currentSubscription.url == requestedSubscription.url) {
+                isNewSubscription = false;
+                break;
+            }
+        }
+
+        if (isNewSubscription)
+            m_cookieChangeSubscriptions.append(WTFMove(requestedSubscription));
+    }
+}
+
+void SWServerRegistration::removeCookieChangeSubscriptions(Vector<CookieChangeSubscription>&& subscriptions)
+{
+    for (auto& requestedSubscription : subscriptions) {
+        size_t i = 0;
+        for (i = 0; i < m_cookieChangeSubscriptions.size(); i++) {
```

Comment:
Seems like this could use `Vector::removeFirstMatching()` but also, not sure Vector is the right data structure for the job.

Reply:
Changed to use HashSet.